tests/lib/rules/func-name-matching.js

```diff
@@ -0,0 +1,107 @@
+/**
+ * @fileoverview Tests for func-name-matching rule.
+ * @author Annie Zhang
+ */
+
+"use strict";
+
+//------------------------------------------------------------------------------
+// Requirements
+//------------------------------------------------------------------------------
+
+const rule = require("../../../lib/rules/func-name-matching"),
+    RuleTester = require("../../../lib/testers/rule-tester");
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester();
+
+ruleTester.run("func-name-matching", rule, {
+    valid: [
+        { code: "var foo = function foo() {};" },
+        { code: "var foo = function() {}"},
+        { code: "foo = function foo() {};" },
+        { code: "obj.foo = function foo() {};" },
+        { code: "obj.foo = function() {};" },
+        { code: "obj.bar.foo = function foo() {};" },
+        { code: "obj['foo'] = function foo() {};" },
+        { code: "var obj = {foo: function foo() {}};" },
+        { code: "var obj = {foo: function() {}};" },
+        { code: "var obj = {['x' + 2]: function bar(){}};", parserOptions: { ecmaVersion: 6} },
+        { code: "obj['x' + 2] = function bar(){};" },
```

Comment:
Could we have a test with just a computed property (no literal involved)? Something like `{ code: "obj[foo] = function bar(){};" }`